packages/playwright-core/src/server/launchApp.ts

```diff
@@ -32,14 +32,18 @@ export async function launchApp(browserType: BrowserType, options: {
   sdkLanguage: string,
   windowSize: types.Size,
   windowPosition?: types.Point,
-  persistentContextOptions?: Parameters<BrowserType['launchPersistentContext']>[2];
+  persistentContextOptions?: Parameters<BrowserType['launchPersistentContext']>[2],
+  initialUrl?: string,
 }) {
   const args = [...options.persistentContextOptions?.args ?? []];
 
   let channel = options.persistentContextOptions?.channel;
+  let hasInitialUrl = false;
   if (browserType.name() === 'chromium') {
+    const initialUrl = options.initialUrl ?? 'data:text/html,';
+    hasInitialUrl = options.initialUrl !== undefined;
     args.push(
-        '--app=data:text/html,',
+        `--app=${initialUrl}`,
```

Comment:
Tests are failing, perhaps because we want to set things up before navigating to the page? For example, bindings or init scripts.